src/modules/keycloak-administration/domain/keycloak-user.service.spec.ts

```diff
@@ -1509,4 +1509,77 @@ describe('KeycloakUserService', () => {
             });
         });
     });
+
+    describe('updateUsername', () => {
+        let username: string;
+        let newUsername: string;
+
+        beforeEach(() => {
+            username = faker.internet.userName();
+            newUsername = faker.internet.userName();
+        });
+
+        describe('when user could not be updated', () => {
+            it('should return error result', async () => {
+                kcUsersMock.find.mockResolvedValueOnce([
+                    {
+                        username: faker.string.alphanumeric(),
+                        email: faker.internet.email(),
+                        id: faker.string.uuid(),
+                        createdTimestamp: faker.date.recent().getTime(),
+                    },
+                ]);
+
+                kcUsersMock.update.mockRejectedValueOnce(new Error());
+
+                const res: Result<void, DomainError> = await service.updateUsername(username, newUsername);
+
+                expect(res.ok).toBeFalsy();
+            });
+        });
+
+        describe('when updating user is successful', () => {
+            it('should log info', async () => {
+                kcUsersMock.find.mockResolvedValueOnce([
+                    {
+                        username: faker.string.alphanumeric(),
+                        email: faker.internet.email(),
+                        id: faker.string.uuid(),
+                        createdTimestamp: faker.date.recent().getTime(),
+                    },
+                ]);
+
+                kcUsersMock.update.mockResolvedValueOnce();
+
+                const res: Result<void, DomainError> = await service.updateUsername(username, newUsername);
+
+                expect(res.ok).toBeTruthy();
+            });
+        });
+
+        describe('when user does not exist', () => {
+            it('should return error', async () => {
+                kcUsersMock.find.mockRejectedValueOnce(new Error());
+
+                const res: Result<void, DomainError> = await service.updateUsername(username, newUsername);
+
+                expect(res.ok).toBeFalsy();
+            });
+        });
+
+        describe('when getAuthedKcAdminClient fails', () => {
+            it('should pass along error result', async () => {
+                const error: Result<KeycloakAdminClient, DomainError> = {
+                    ok: false,
+                    error: new KeycloakClientError('Could not authenticate'),
+                };
+
+                adminService.getAuthedKcAdminClient.mockResolvedValueOnce(error);
+
+                const res: Result<void, DomainError> = await service.updateUsername(username, newUsername);
+
+                expect(res).toBe(error);
+            });
+        });
+    });
```

Comment:
These tests don't really tests anything. At least add an expect to make sure, that `kcUsersMock.update` was actually called with the correct parameters

Reply:
We are testing the kcService and not the kcUsersMock which is just beeing set up.